small/mempool.h

```diff
@@ -261,23 +274,37 @@ void
 mslab_free(struct mempool *pool, struct mslab *slab, void *ptr);
 
 /**
- * Free a single object.
- * @pre the object is allocated in this pool.
+ * Helper function for quick free up memory. In case we know
+ * slab we don't need to find it from ptr. Used in case when
+ * mempool is a part of small_alloc. We find slab from ptr in
+ * smfree function and we don't need to search for it again to
+ * free up memory.
  */
 static inline void
-mempool_free(struct mempool *pool, void *ptr)
+mempool_free_slab(struct mempool *pool, struct mslab *slab, void *ptr)
```

Comment:
Hm, mslab now has pointer to its mempool, why do we need then pass mempool to free slab?
Anyway assert(slab->mempool == pool) would be suitable